backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewCreateService.java

```diff
@@ -239,4 +240,25 @@ public static boolean isEmptyObjectiveRequirement(Element self) {
                 .map(ObjectiveMembership.class::cast)
                 .findFirst().isEmpty();
     }
+
+    public Element createPartUsageAsAllocationDefinitionEnd(Element self, Element endParent) {
+        if (self instanceof AllocationDefinition) {
+            // create the part usage that is used as the end element
+            PartUsage newPartUsage = SysmlFactory.eINSTANCE.createPartUsage();
+            newPartUsage.setDeclaredName(self.getDeclaredName() + "'s end");
+            var membership = SysmlFactory.eINSTANCE.createOwningMembership();
+            membership.getOwnedRelatedElement().add(newPartUsage);
+            endParent.getOwnedRelationship().add(membership);
+            // create end model tree
```

Comment:
what is an end model tree here?

Reply:
This is the FeatureMembership that contains the ReferenceUsage containing the FeatureTyping containing the PartUsage previously created. 
I admit that this comment is not clear at all. I will remove it